src/sentry/workflow_engine/handlers/detector/n_plus_one_api_calls_detector_handler.py

```diff
@@ -0,0 +1,375 @@
+from __future__ import annotations
+
+import hashlib
+import os
+from collections import defaultdict
+from collections.abc import Mapping, Sequence
+from datetime import timedelta
+from typing import Any
+from urllib.parse import parse_qs, urlparse
+
+from django.utils.encoding import force_bytes
+
+from sentry.issues.grouptype import PerformanceNPlusOneAPICallsGroupType
+from sentry.issues.issue_occurrence import IssueEvidence
+from sentry.models.organization import Organization
+from sentry.models.project import Project
+from sentry.utils.performance_issues.base import (
+    DetectorType,
+    fingerprint_http_spans,
+    get_notification_attachment_body,
+    get_span_evidence_value,
+    get_url_from_span,
+    parameterize_url,
+)
+from sentry.utils.performance_issues.detectors.utils import get_total_span_duration
+from sentry.utils.performance_issues.performance_problem import PerformanceProblem
+from sentry.utils.performance_issues.types import PerformanceProblemsMap, Span
+from sentry.workflow_engine.handlers.detector.base import DetectorEvaluationResult
+from sentry.workflow_engine.handlers.detector.performance_detector_handler import (
+    PerformanceDetectorHandler,
+)
+from sentry.workflow_engine.models import DataPacket
+from sentry.workflow_engine.types import DetectorGroupKey, DetectorPriorityLevel
+
+
+class NPlusOneAPICallsDetectorHandler(PerformanceDetectorHandler):
+    __slots__ = ["stored_problems"]
+    type = DetectorType.N_PLUS_ONE_API_CALLS
+    settings_key = DetectorType.N_PLUS_ONE_API_CALLS
+
+    HOST_DENYLIST: list[str] = []
+
+    def __init__(self, settings: dict[DetectorType, Any], event: dict[str, Any]) -> None:
+        super().__init__(settings, event)
+        # init with a Detector
+
+        # TODO: Only store the span IDs and timestamps instead of entire span objects
+        self.stored_problems: PerformanceProblemsMap = {}
+        self.spans: list[Span] = []
+        self.span_hashes: dict[str, str | None] = {}
+
+    def evaluate(self, data_packet: DataPacket) -> dict[DetectorGroupKey, DetectorEvaluationResult]:
```

Comment:
This `DataPacket` should have a type associated with it. The type could just be associated with `NPlusOneAPICallsDetectorHandler` or `PerformanceDetectorHandler` (since I guess it's always a span? or a list of spans? idk)